database/testing/suite.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package testing
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"math/rand"
+	"net"
+	"strconv"
+
+	"github.com/juju/testing"
+
+	"github.com/canonical/go-dqlite/app"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+// DBSuite is used to provider a Dqlite-backed sql.DB reference to tests.
+type DBSuite struct {
+	testing.IsolationSuite
+
+	dqlite *app.App
+	db     *sql.DB
+}
+
+// SetUpSuite creates a new Dqlite application and waits for it to be ready.
+func (s *DBSuite) SetUpSuite(c *gc.C) {
+	s.IsolationSuite.SetUpSuite(c)
+
+	dbPath := c.MkDir()
+	port := findTCPPort(c)
```

Comment:
I wish we had a better way to do this, by telling dqlite to launch on a random port and tell us what port it is on. But certainly this is what we've done for mongo, etc tests.

Reply:
Indeed, this was copied from the Mongo testing code.